video/video.ino

```diff
@@ -218,13 +223,43 @@ void setConsoleMode(bool mode) {
 
 // Switch to terminal mode
 //
-void switchTerminalMode() {
+void switchTerminalMode(bool enable) {
+	if (terminalMode == enable) {
+		return;
+	}
+	exitTerminal = false;
 	cls(true);
 	canvas.reset();
-	Terminal.begin(_VGAController.get());	
-	Terminal.connectSerialPort(VDPSerial);
-	Terminal.enableCursor(true);
-	terminalMode = true;
+	if (enable) {
+		Terminal = std::unique_ptr<fabgl::Terminal>(new fabgl::Terminal());
+		Terminal->begin(_VGAController.get());	
+		Terminal->connectSerialPort(VDPSerial);
+		Terminal->enableCursor(true);
+		// onVirtualKey is triggered whenever a key is pressed or released
+		Terminal->onVirtualKeyItem = [&](VirtualKeyItem * vkItem) {
+			if (vkItem->vk == VirtualKey::VK_F12) {
+				if (vkItem->CTRL && (vkItem->LALT || vkItem->RALT)) {
+					// CTRL + ALT + F12: emergency exit terminal mode
+					exitTerminal = true;
+				}
+			}
+		};
+
+		// onUserSequence is triggered whenever a User Sequence has been received (ESC + '_#' ... '$'), where '...' is sent here
+		Terminal->onUserSequence = [&](char const * seq) {
+			// 'Q!': exit terminal mode
+			if (strcmp("Q!", seq) == 0) {

```

Comment:
this is also an arbitrary escape sequence...  if anyone has any better ideas then fire away :grin: